src/logger.cc

```diff
@@ -110,21 +110,23 @@ NAN_METHOD(Logger::New) {
 
         if (!logger) {
 #if defined(_WIN32)
-          std::wstring_convert<std::codecvt_utf8_utf16<wchar_t>> converter;
-          const std::wstring fileName =
-              converter.from_bytes(*Nan::Utf8String(info[2]));
+          const std::string utf8Filename = *Nan::Utf8String(info[2]);
+          const int bufferLen = MultiByteToWideChar(CP_UTF8, 0, utf8Filename.c_str(), -1, NULL, 0);
+          std::wstring fileName;
+          fileName.resize(bufferLen);
```

Comment:
I tested this change locally and it doesn't work, because spdlog reports "Uncaught Error: Failed opening file  for writing: permission denied" in the test cases. I tried using `GetLastError` to see whether the error comes from the `MultiByteToWideChar` call itself, but that doesn't seem to be the case.

I'm also following https://learn.microsoft.com/en-us/windows/win32/api/stringapiset/nf-stringapiset-multibytetowidechar#code-example.

Reply:
What was different ? Did the filename values get corrupted ?

Reply:
`reserve` changes the capacity but doesn't seem to change the size, so the filename does get corrupted by the time `deps/spdlog` tries to open it.

Example using reserve:
```
Printing filename: C:\Users\raymondzhao\work\native-modules\node-spdlog\test\logs\test.log
Printing filename size: 0
Printing filename capacity: 79
Passed create_dir
fname after create_dir: C:\Users
Printing fname size: 0
Printing fname capacity: 7
```

Example using resize:
```
Printing filename: C:\Users\raymondzhao\work\native-modules\node-spdlog\test\logs\test.log
Printing filename size: 72
Printing filename capacity: 79
Passed create_dir
fname after create_dir: C:\Users\raymondzhao\work\native-modules\node-spdlog\test\logs\test.log      
Printing fname size: 72
Printing fname capacity: 79
fname: C:\Users\raymondzhao\work\native-modules\node-spdlog\test\logs\test.log
```

Reply:
Makes sense, `MultiByteToWideChar` does not append so providing a pointer to the string without initialized size causes incorrect output.